include/exadg/aero_acoustic/calculators/source_term_calculator.h

```diff
@@ -153,6 +154,73 @@ class SourceTermCalculator
     }
   }
 
+  void
+  compute_source_term(dealii::MatrixFree<dim, Number> const &       matrix_free_in,
+                      VectorType &                                  dst,
+                      dealii::Function<dim> const &                 analytical_source_term,
+                      std::pair<unsigned int, unsigned int> const & cell_range) const
+  {
+    CellIntegratorScalar dpdt(matrix_free_in, data.dof_index_pressure, data.quad_index);
+
+    Number rho = static_cast<Number>(data.density);
+
+    auto apply_scaling = get_scaling_function();
+
+    for(unsigned int cell = cell_range.first; cell < cell_range.second; ++cell)
+    {
+      dpdt.reinit(cell);
+
+      for(unsigned int q = 0; q < dpdt.n_q_points; ++q)
+      {
+        scalar flux = -rho * FunctionEvaluator<0, dim, Number>::value(analytical_source_term,
+                                                                      dpdt.quadrature_point(q));
+
+        if(data.blend_in)
+          apply_scaling(flux, dpdt.quadrature_point(q));
```

Comment:
Is there a reason why you do not write this code like `flux *= get_scaling_factor(qpoint);`. Right now I could imagine this would simplify the code (why a lambda function; why two function arguments even though the first one (the flux) seems to be passed to this function only to be multiplied by a scaling factor?) and make the code easier to understand.

Reply:
[2980d74](https://github.com/exadg/exadg/pull/659/commits/2980d74bc7b7357a560a7688772160c3078d18b2)